handler/process/adaptor.go

```diff
@@ -88,6 +88,11 @@ func (a *Adaptor) HandleMessage(
 		return nil
 	}
 
+	if exists && inst.HasEnded {
```

Comment:
```suggestion
	if inst.HasEnded {
```

Pretty sure this can be simplified and the `if` above this one can then be removed.

Reply:
Also, I don't think there's any existing *test* for this logic, and certainly not for events specifically.

Reply:
Can't remove the if block above without breaking a timeout test for a not yet existing instance.

Reply:
I've removed the `exists` check from that line and added another test to verify messages to an already ended process get ignored.